src/vsr/replica.zig

```diff
@@ -2242,18 +2340,220 @@ pub fn ReplicaType(
             }
         }
 
+        fn on_request_sync_manifest(self: *Self, message: *Message) void {
+            assert(message.header.command == .request_sync_manifest);
+            if (self.ignore_sync_request_message(message)) return;
+
+            const reply = self.message_bus.get_message();
+            defer self.message_bus.unref(reply);
+
+            const offset = message.header.request;
+            const size = self.superblock.staging.manifest_size;
+            stdx.copy_disjoint(
+                .inexact,
+                u8,
+                reply.buffer[@sizeOf(Header)..],
+                self.superblock.manifest_buffer[offset..][0 .. size - offset],
+            );
+
+            reply.header.* = .{
+                .command = .sync_manifest,
+                .cluster = self.cluster,
+                .replica = self.replica,
+                .size = @sizeOf(Header) + (size - offset),
```

Comment:
I am confused here --- we don't check whether `size` fits `message_body_size_max`, but, if we always can send the whole thing in a single message, then why do we need offsets?